src/mlpack/methods/ann/models/yolov3/yolov3_layer_impl.hpp

```diff
@@ -0,0 +1,254 @@
+/**
+ * @file methods/ann/models/yolov3/yolov3_layer_impl.hpp
+ * @author Andrew Furey
+ *
+ * Implementation of the YOLOv3 layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_MODELS_YOLOV3_YOLOV3_LAYER_IMPL_HPP
+#define MLPACK_METHODS_ANN_MODELS_YOLOV3_YOLOV3_LAYER_IMPL_HPP
+
+#include "yolov3_layer.hpp"
+
+namespace mlpack {
+
+template <typename MatType>
+YOLOv3Layer<MatType>::YOLOv3Layer(
+    const size_t imgSize,
+    const size_t numAttributes,
+    const size_t gridSize,
+    const size_t predictionsPerCell,
+    const std::vector<typename MatType::elem_type> anchors) :
+    Layer<MatType>(),
+    imgSize(imgSize),
+    numAttributes(numAttributes),
+    gridSize(gridSize),
+    grid(gridSize * gridSize),
+    predictionsPerCell(predictionsPerCell)
+{
+  if (anchors.size() != 2 * predictionsPerCell)
+  {
+    std::ostringstream errMessage;
+    errMessage << "YOLOv3 must have " << predictionsPerCell
+                << " (w, h) anchors but you gave "
+                << anchors.size() / 2 << ".";
+    throw std::logic_error(errMessage.str());
+  }
+
+  anchorsW = MatType(grid, predictionsPerCell, arma::fill::none);
+  anchorsH = MatType(grid, predictionsPerCell, arma::fill::none);
+
+  // Could maybe use .each_row()?
```

Comment:
I think that bandicoot supports `each_row()` in the way you need... I believe it could work?